### ISSUE TITLE
Flakey Infinite Progress Bar Tests

### ISSUE THREAD

**Issue by stuartmscott** [fyne-io/fyne#874]
Running the widget tests multiple times caused panic:

> go test -count=100 fyne.io/fyne/widget/...

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x12d3063]

goroutine 3789 [running]:
sync.(*RWMutex).RLock(...)
	/usr/local/go/src/sync/rwmutex.go:48
fyne.io/fyne/test.(*testDriver).CanvasForObject(0x0, 0x14d3d20, 0xc0007b32c0, 0x0, 0x0)
	/Users/stuartscott/Documents/Projects/Go/src/fyne.io/fyne/test/testdriver.go:30 +0x33
fyne.io/fyne/canvas.Refresh(0x14d3d20, 0xc0007b32c0)
	/Users/stuartscott/Documents/Projects/Go/src/fyne.io/fyne/canvas/base.go:69 +0x81
fyne.io/fyne/widget.(*infProgressRenderer).Refresh(0xc0007b3300)
	/Users/stuartscott/Documents/Projects/Go/src/fyne.io/fyne/widget/progressbarinfinite.go:91 +0x6c
fyne.io/fyne/widget.(*infProgressRenderer).infiniteProgressLoop(0xc0007b3300)
	/Users/stuartscott/Documents/Projects/Go/src/fyne.io/fyne/widget/progressbarinfinite.go:115 +0x48
created by fyne.io/fyne/widget.(*infProgressRenderer).start
	/Users/stuartscott/Documents/Projects/Go/src/fyne.io/fyne/widget/progressbarinfinite.go:100 +0xe9
FAIL	fyne.io/fyne/widget	31.271s
FAIL
```

**Comment by stuartmscott** [comment]
Upon closer inspection it looks like a duplicate of #854

**Comment by andydotxyz** [comment]
Yes, this appears resolved by dbc25b09
